### PR TITLE
openxla pin update to 06/05

### PR DESCRIPTION
**Pull request by zpcore**
NIT: Also relocate the commented `local_repository` right after http_archive or it will report xla not found.

### CODE REVIEW

**Review by ManfeiBai — APPROVED** [review]
Thanks, LGTM, please merge after all CI green

File: WORKSPACE [hunk 1]
```diff
@@ -70,6 +70,19 @@ http_archive(
     ],
 )
 
+# For development, one often wants to make changes to the OpenXLA repository as well
```
Comment (JackCaoG) [inline WORKSPACE:73]: why is this moved?

Comment (zpcore) [reply WORKSPACE:73]: local_repository needs to be stated before `load("@xla//third_party/py:python_init_rules.bzl", "python_init_rules")...` etc. Those are newly added from hermetic python.
